internal/controlplane/handlers_projects.go

```diff
@@ -201,6 +201,15 @@ func (s *Server) CreateProject(
 		return nil, status.Errorf(codes.Internal, "error creating subproject: %v", err)
 	}
 
+	// Retrieve the membership-to-feature mapping from the configuration
+	projectFeatures := s.cfg.Features.GetFeaturesForMemberships(ctx)
+	if err := qtx.CreateEntitlements(ctx, db.CreateEntitlementsParams{
+		Features:  projectFeatures,
+		ProjectID: subProject.ID,
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating entitlements: %v", err)
+	}
```

Comment:
This is sort-of-weird: this is creating a child project free-form here, separate from the `internal/projects` code.  It's not clear to me whether these child projects should:

1. Copy entitlements from the parent project
2. Use entitlements based on the specific parent project member who creates them
3. Have an empty set of entitlements, and the entitlement queries are extended to check parents.

Right now, it looks like were doing option 2, but it's not clear that is correct.

Let's file an issue and assign it to @ethomson to figure out the desired way to handle entitlements in sub-projects, because those are currently behind an entitlement, and we _probably_ know most of the users of that feature.

Reply:
Also, file a bug to move this code to `internal/projects`, so all project creation happens in that one module, and `internal/controlplane` gets a bit smaller.

Reply:
I did find it weird that we have two separate paths for project creation, especially one of them being exclusive to the handler and was going to follow-up on that, thanks for noting it as well.

Here are the new issues:

https://github.com/mindersec/minder/issues/5001
https://github.com/mindersec/minder/issues/5002